#
#
# patch "libpurple/protocols/msnp9/notification.c"
#  from [dd690580185c00b68db4240fd22732b5e3770c36]
#    to [83db9e1346bfbee5a95f55cfe10508d78fdac196]
#
============================================================
--- libpurple/protocols/msnp9/notification.c	dd690580185c00b68db4240fd22732b5e3770c36
+++ libpurple/protocols/msnp9/notification.c	83db9e1346bfbee5a95f55cfe10508d78fdac196
@@ -780,14 +780,27 @@ rea_cmd(MsnCmdProc *cmdproc, MsnCommand
 static void
 rea_cmd(MsnCmdProc *cmdproc, MsnCommand *cmd)
 {
-	/* TODO: This might be for us too */
-
 	MsnSession *session;
+	PurpleAccount *account;
 	PurpleConnection *gc;
 	const char *friendly;
+	char *username;

 	session = cmdproc->session;
-	gc = session->account->gc;
+	account = session->account;
+	username = g_strdup(purple_normalize(account,
+						purple_account_get_username(account)));
+
+	/* Only set display name if our *own* friendly name changed! */
+	if (strcmp(username, purple_normalize(account, cmd->params[2])))
+	{
+		g_free(username);
+		return;
+	}
+
+	g_free(username);
+
+	gc = account->gc;
 	friendly = purple_url_decode(cmd->params[3]);

 	purple_connection_set_display_name(gc, friendly);
